plugins/api-docs/README.md

```diff
@@ -85,6 +85,35 @@ security:
     - [read_pets, write_pets]
 ```
 
+### Customizing table columns and filters
+
+It's possible to specify which columns and filters you want this plugin to
+display. You can do so by declaring the columns and/or filters as props for the
+`ApiExplorerPage` component like so:
+
+```tsx
+const domainColumn = { title: 'Domain', field: 'entity.metadata.domain' };
+// We can use plain strings to denote default Backstage columns...
+const customColumns = [
+  'Name',
+  'Description',
+  'Owner',
+  'Lifecycle',
+  'Type',
+  domainColumn,
+];
```

Comment:
I wonder if a pattern similar to https://github.com/backstage/backstage/blob/master/plugins/catalog-react/src/components/EntityTable/EntityTable.tsx#L29 would be a good choice.

Beside that, I wonder if we really want to make the table customizable. In the end the causes a lot of complexity that could be avoided if we provide the building blocks to create a new page with a table instead that replaces this page in the individual customer apps. Otherwise we have to repeat the customization stuff for every view?

Reply:
I had no idea this `EntityTable` already existed as I've never come across it. I agree this seems like a nice pattern to use here. The downside being adding or removing existing columns may require replacing all columns. However, I think there is an opportunity here to focus on a column composition API that could in theory be re-used across all tables that use the EntityTable or a similar pattern.

Reply:
Comment by @freben https://github.com/backstage/backstage/issues/4757#issuecomment-803918721 might also be an option here.

Reply:
Ah, so there is a pattern taking shape around this. Cool. I'm also in agreement that going the composition route would yield a more robust solution for this type of customization. I'll have another go at this, bearing in mind that `EntityTable` pattern and the suggestions from @freben's comment.